k8s-plugin-keel/src/main/kotlin/com/amazon/spinnaker/keel/k8s/verificationEvaluator/K8sJobEvaluator.kt

```diff
@@ -0,0 +1,182 @@
+// Copyright 2021 Amazon.com, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.amazon.spinnaker.keel.k8s.verificationEvaluator
+
+import com.amazon.spinnaker.keel.k8s.*
+import com.amazon.spinnaker.keel.k8s.model.K8sJobVerification
+import com.netflix.spinnaker.keel.api.ArtifactInEnvironmentContext
+import com.netflix.spinnaker.keel.api.Verification
+import com.netflix.spinnaker.keel.api.action.ActionState
+import com.netflix.spinnaker.keel.api.actuation.SubjectType.VERIFICATION
+import com.netflix.spinnaker.keel.api.actuation.TaskLauncher
+import com.netflix.spinnaker.keel.api.constraints.ConstraintStatus
+import com.netflix.spinnaker.keel.api.plugins.VerificationEvaluator
+import com.netflix.spinnaker.keel.model.OrcaJob
+import com.netflix.spinnaker.keel.orca.OrcaService
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
+import org.slf4j.LoggerFactory
+import retrofit2.HttpException
+
+class K8sJobEvaluator(
+    private val taskLauncher: TaskLauncher,
+    private val orcaService: OrcaService
+) : VerificationEvaluator<K8sJobVerification> {
+    private val log by lazy { LoggerFactory.getLogger(javaClass) }
+    override val supportedVerification =
+        VERIFICATION_K8S_JOB_V1 to K8sJobVerification::class.java
+
+    override fun evaluate(
+        context: ArtifactInEnvironmentContext,
+        verification: Verification,
+        oldState: ActionState
+    ): ActionState {
+        val taskId = oldState.metadata["taskId"]
+        requireNotNull(taskId) {
+            "task ID not found in previous state. please check last K8s job executed by this verifyWith: $verification"
```

Comment:
can we return `taskId` as part of the error message too?

Reply:
yep